client/lib/importer/actions.js

```diff
@@ -2,20 +2,50 @@
  * External dependencies
  */
 import Dispatcher from 'dispatcher';
+import noop from 'lodash/utility/noop';
 const wpcom = require( 'lib/wp' ).undocumented();
 
 /**
  * Internal dependencies
  */
-import { actionTypes } from './constants';
+import { actionTypes, appStates } from './constants';
 import { fromApi, toApi } from './common';
 
-export function cancelImport( siteId, importerId ) {
+const ID_GENERATOR_PREFIX = 'local-generated-id-';
+
+const cancelOrder = ( siteId, importerId ) => toApi( { importerId, importerState: appStates.CANCEL_PENDING, site: { ID: siteId } } );
+const apiStart = () => Dispatcher.handleViewAction( { type: actionTypes.API_REQUEST } );
+const apiSuccess = () => Dispatcher.handleViewAction( { type: actionTypes.API_SUCCESS } );
+const apiFailure = () => Dispatcher.handleViewAction( { type: actionTypes.API_FAILURE } );
+
+function apiUpdateImporter( importerStatus ) {
+	apiSuccess();
+
+	Dispatcher.handleViewAction( {
+		type: actionTypes.RECEIVE_IMPORT_STATUS,
+		importerStatus
+	} );
+}
+
+export function cancelImport( siteId, importerId, uploadAborter = noop ) {
+	uploadAborter();
+
 	Dispatcher.handleViewAction( {
 		type: actionTypes.CANCEL_IMPORT,
 		importerId,
 		siteId
 	} );
+
+	if ( importerId.includes( ID_GENERATOR_PREFIX ) ) {
```

Comment:
>Did you have some test suite at the time or is there any way to know if this could have changed since then or if Babel 6 would make a difference?

To my knowledge, we've only found out the hard way (broken screens in production).

I did a bit of digging, and we may need to add the [`babel-polyfill` module](http://babeljs.io/docs/usage/polyfill/) if we want to start using these methods.

We use the [runtime module](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-runtime) which, as far as I understand, transforms calls to ES6 base object methods (e.g. `Array.from`, `Object.assign`) to their `core-js` equivalents, which is why we're able to use those.